game/src/contrib/crafting/Crafting.java

```diff
@@ -0,0 +1,185 @@
+package contrib.crafting;
+
+import com.badlogic.gdx.utils.JsonReader;
+import com.badlogic.gdx.utils.JsonValue;
+
+import contrib.crafting.ingredient.CraftingIngredient;
+import contrib.crafting.ingredient.CraftingItemIngredient;
+import contrib.crafting.result.CraftingItemResult;
+import contrib.crafting.result.CraftingResult;
+
+import starter.Main;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.logging.Logger;
+
+public class Crafting {
+
+    private static final HashSet<Recipe> recipes = new HashSet<>();
+    private static final Logger logger = Logger.getLogger(Crafting.class.getName());
+
+    /**
+     * Get a recipe based on the provided items.
+     *
+     * <p>If there are multiple recipes that can be made with the provided items, recipes where
+     * items must be in a specific order will be prioritized.
+     *
+     * @param inputs Ingredients used.
+     * @return The recipe that can be crafted with the provided ingredients. If none can be crafted,
+     *     the returned optional will be empty.
+     */
+    public static Optional<Recipe> getRecipeByIngredients(CraftingIngredient[] inputs) {
+
+        List<Recipe> possibleRecipes = new ArrayList<>();
+        for (Recipe recipe : recipes) {
+            if (recipe.canCraft(inputs)) {
+                possibleRecipes.add(recipe);
+            }
+        }
+
+        if (possibleRecipes.isEmpty()) {
+            return Optional.empty();
+        }
+        if (possibleRecipes.size() == 1) {
+            return Optional.of(possibleRecipes.get(0));
+        }
+        possibleRecipes.sort(Comparator.comparing(c -> !c.isOrdered()));
+
+        return Optional.of(possibleRecipes.get(0));
+    }
+
+    public static void addRecipe(Recipe recipe) {
+        if (recipe.getIngredients().length == 0) {
+            throw new InvalidRecipeException("Recipes with no ingredients are not allowed!");
+        }
+        recipes.add(recipe);
+    }
+
+    public static void removeRecipe(Recipe recipe) {
+        recipes.remove(recipe);
+    }
+
+    public static void clearRecipes() {
+        recipes.clear();
+    }
+
+    /**
+     * Load recipes from the recipes folder.
+     *
+     * <p>If the program is compiled to a jar file, recipes will be loaded from within the jar file.
+     */
+    public static void loadRecipes() {
+        if (Objects.requireNonNull(Crafting.class.getResource("/recipes"))
+                .toString()
+                .startsWith("jar:")) {
+            loadFromJar();
+        } else {
+            loadFromFile();
+        }
+    }
+
+    /** Load recipes if the program was started from a jar file. */
+    private static void loadFromJar() {
```

Comment:
@malt-r @JudiTeller ihr wolltet doch die Tage irgendwas laden und hattet bedenken bzgl. der Funktionalität in einer jar.
Hilft das hier irgendwie?

Reply:
In einen der nächsten Commits habe ich die exportierte Jar noch ausführbar gemacht, was sie aus irgend einem Grund nicht 🤷‍♂️

Reply:
Das ist genau das, was in #695 benötigt wird.
Ich denke mal, dass das nicht die einzigen zwei Stellen sind, bei denen unterschiedlich gesucht werden muss.
Sollten wir die Funktionalität dann nicht in eine eigene Klasse herausziehen?

Reply:
Klingt sinnvoll.
@JudiTeller erstellest du dafür ein passendes Issue und linkst auf diese Diskussion/ stelle im Code.
@fwatermann du kannst das hier erstmal so lassen und wir extrahieren das danach. Dieser PR muss ja mal gemerged werden